TeamCode/src/main/java/org/firstinspires/ftc/teamcode/auton/autonPark.java

```diff
@@ -0,0 +1,130 @@
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "rightRear", "setDirection:FORWARD"));
+
+        // Init arm and Viper Slide DcMotors.
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "armJoint", "setMode:RUN_USING_ENCODER"));
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "leftViperSlide", "setMode:RUN_USING_ENCODER"));
+        robot.introduce(new HardwareElement<>(DcMotor.class, hardwareMap, "rightViperSlide", "setMode:RUN_USING_ENCODER"));
+
+        telemetry.addData("Status", "Initialized");
+    }
+
+    @Override
+    public void init_loop() {
+        telemetry.addLine("Y: Backboard Blue \n A: Wall Blue \n X: Backboard Red \n" +
+                "B: Wall Red");
+        telemetry.addLine("***");
+        telemetry.addLine("Right Bumper: Corner Parking \n Left Bumper: Middle Parking");
+
+        updateStartingPosition();
+        updateParkingPosition();
+    }
+
+    private void updateStartingPosition() {
+        if (gamepad1.y || startingLocationIndex == 0) {
+            startingLocationIndex = 0;
+            telemetry.addData("Starting Position", "Backboard Blue");
+        }
+
+        if (gamepad1.a || startingLocationIndex == 1) {
+            startingLocationIndex = 1;
+            telemetry.addData("Starting Position", "Wall Blue");
+        }
+
+        if (gamepad1.x || startingLocationIndex == 2) {
+            startingLocationIndex = 2;
+            telemetry.addData("Starting Position", "Backboard Red");
+        }
+
+        if (gamepad1.b || startingLocationIndex == 3) {
+            startingLocationIndex = 3;
+            telemetry.addData("Starting Position", "Wall Red");
+        }
+    }
+
+    private void updateParkingPosition() {
+        if (gamepad1.right_bumper && startingLocationIndex <= 1 || parkingPositionIndex == 0) {
+            parkingPositionIndex = 0;
+            telemetry.addData("Parking Position", "Blue Corner");
+        }
+
+        if (gamepad1.right_bumper || parkingPositionIndex == 0){
+            parkingPositionIndex = 1;
+            telemetry.addData("Parking Position", "Red Corner");
+        }
+
+        if (gamepad1.left_bumper && startingLocationIndex <= 1 || parkingPositionIndex == 2) {
+            parkingPositionIndex = 2;
+            telemetry.addData("Parking Position", "Blue Middle");
+        }
+
+        if (gamepad1.left_bumper || parkingPositionIndex == 3) {
+            parkingPositionIndex = 3;
+            telemetry.addData("Parking Position", "Red Middle");
+        }
+    }
+
+    @Override
+    public void start() {
+        handleStartPosition();
+        RobotMethods.driveTo(robot, context, parkingPosition[parkingPositionIndex][0],
+                parkingPosition[parkingPositionIndex][1], motorRunPower, tolerance, telemetry);
+    }
+
+    @Override
+    public void loop() {
+        telemetry.addLine("Uh oh");
```

Comment:
Is there a point to this, cause it might mess up the telemetry by spamming it messages.

Reply:
Shoot lol i forgot to get rid of that